backend-diagram-converter/core/src/main/java/org/camunda/community/converter/BpmnDiagramCheckResult.java

```diff
@@ -4,13 +4,20 @@
 import java.util.List;
 
 public class BpmnDiagramCheckResult {
-  private String filename;
   private final List<BpmnElementCheckResult> results = new ArrayList<>();
+  private String filename;
 
   public List<BpmnElementCheckResult> getResults() {
     return results;
   }
 
+  public BpmnElementCheckResult getResult(String elementId) {
```

Comment:
Do we need this method here? I would try to keep all Pojos as clean as possible (only simple getter and setter).

Maybe we could extract it to a helper class.

Reply:
I once had this as a map. Then, I decided that it should be a list. The reason was back then that iterating a list is way easier in js as iterating a map.

So, we can change it which will cause some refactoring.

If you agree, I would open a ticket for this and we leave the method as it is in the meantime.

Reply:
We should keep both methods `List<BpmnElementCheckResult> getResults()` and `BpmnElementCheckResult getResult(String elementId)`.

Refactoring can be done in this ticket as only the internals change and the tests are stable.